simple_pe/simple_pe_magma.py

```diff
@@ -0,0 +1,21 @@
+import magma as m
+import mantle
+
+
+def define_pe(ops, T=m.UInt, data_width=16):
+    T = T(data_width)
+    opcode_width = m.bitutils.clog2(len(ops))
+
+    class PE(m.Circuit):
+        name = "PE"
+        IO = ["opcode", m.In(m.Bits(opcode_width)), "I0", m.In(T), "I1",
+              m.In(T), "O", m.Out(T)]
+
+        @classmethod
+        def definition(io):
```

Comment:
can we not use the new N-bit mux from mantle that doesn't require powers of 2? Just to abstract away the implementation of the mux (could be big mux, chain of muxes, case statement etc.).

Alternatively, it would be cool to use the combinational circuit syntax!

Reply:
Done